VirtualReality/Widgets/qMRMLVirtualRealityDataModuleWidget.cxx

```diff
@@ -0,0 +1,111 @@
+#include "ui_qMRMLVirtualRealityDataModuleWidget.h"
+
+// VirtualReality MRML includes
+#include "vtkMRMLVirtualRealityViewNode.h"
+
+// Qt includes
+#include <QDebug>
+
+//-----------------------------------------------------------------------------
+class qMRMLVirtualRealityDataModuleWidgetPrivate : public Ui_qMRMLVirtualRealityDataModuleWidget
+{
+  Q_DECLARE_PUBLIC(qMRMLVirtualRealityDataModuleWidget);
+
+protected:
+  qMRMLVirtualRealityDataModuleWidget* const q_ptr;
+
+public:
+  qMRMLVirtualRealityDataModuleWidgetPrivate(qMRMLVirtualRealityDataModuleWidget& object);
+  virtual ~qMRMLVirtualRealityDataModuleWidgetPrivate();
+
+  void init();
+
+public:
+  /// Virtual reality view MRML node
+  vtkWeakPointer<vtkMRMLVirtualRealityViewNode> VirtualRealityViewNode;
+};
+
+//-----------------------------------------------------------------------------
+qMRMLVirtualRealityDataModuleWidgetPrivate::qMRMLVirtualRealityDataModuleWidgetPrivate(qMRMLVirtualRealityDataModuleWidget& object)
+  : q_ptr(&object)
+{
+}
+
+//-----------------------------------------------------------------------------
+qMRMLVirtualRealityDataModuleWidgetPrivate::~qMRMLVirtualRealityDataModuleWidgetPrivate()
+{
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityDataModuleWidgetPrivate::init()
+{
+  Q_Q(qMRMLVirtualRealityDataModuleWidget);
+  this->setupUi(q);
+}
+
+//-----------------------------------------------------------------------------
+// qMRMLVirtualRealityDataModuleWidget methods
+
+//-----------------------------------------------------------------------------
+qMRMLVirtualRealityDataModuleWidget::qMRMLVirtualRealityDataModuleWidget(QWidget* _parent)
+  : qMRMLWidget(_parent)
+  , d_ptr(new qMRMLVirtualRealityDataModuleWidgetPrivate(*this))
+{
+  Q_D(qMRMLVirtualRealityDataModuleWidget);
+  d->init();
+
+  this->updateWidgetFromMRML();
+}
+
+//-----------------------------------------------------------------------------
+qMRMLVirtualRealityDataModuleWidget::~qMRMLVirtualRealityDataModuleWidget()
+= default;
+
+//-----------------------------------------------------------------------------
+vtkMRMLVirtualRealityViewNode* qMRMLVirtualRealityDataModuleWidget::virtualRealityViewNode() const
+{
+  Q_D(const qMRMLVirtualRealityDataModuleWidget);
+  return d->VirtualRealityViewNode;
+}
+
+//-----------------------------------------------------------------------------
+QString qMRMLVirtualRealityDataModuleWidget::virtualRealityViewNodeID()const
+{
+  Q_D(const qMRMLVirtualRealityDataModuleWidget);
+  return (d->VirtualRealityViewNode.GetPointer() ? d->VirtualRealityViewNode->GetID() : QString());
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityDataModuleWidget::updateWidgetFromMRML()
+{
+  //Q_D(qMRMLVirtualRealityDataModuleWidget);
+}
+
+qMRMLSubjectHierarchyTreeView* qMRMLVirtualRealityDataModuleWidget::treeView()
```

Comment:
Please add separator that appears above each function.

Also please add an empty new line to the end of the file